npm/webpack-dev-server-fresh/src/helpers/nuxtHandler.ts

```diff
@@ -0,0 +1,26 @@
+import type { CreateFinalWebpackConfig } from '../createWebpackDevServer'
+import debugLib from 'debug'
+import type { Configuration } from 'webpack'
+
+type PresetHandler = Omit<CreateFinalWebpackConfig, 'frameworkConfig'>
+
+const debug = debugLib('cypress:webpack-dev-server-fresh:nuxtHandler')
+
+export async function nuxtHandler ({ devServerConfig }: PresetHandler): Promise<Configuration> {
+  try {
+    const nuxt = require.resolve('nuxt', {
+      paths: [devServerConfig.cypressConfig.projectRoot],
+    })
+
+    const { getWebpackConfig } = require(nuxt)
+
+    const webpackConfig = await getWebpackConfig()
```

Comment:
😢 that this forces async everywhere. Was looking into nuxt internal to understand why it does, and it seems only to be because of the ability to have nuxt options be async: https://github.com/nuxt/nuxt.js/blame/db11f7bf937cdbb96ff0067bdee1aa153978e4e0/packages/config/src/load.js#L63

Oh well, probably not worth tapping into internal APIs, but I was half wondering if we could just source from [@nuxt/webpack](https://github.com/nuxt/nuxt.js/tree/db11f7bf937cdbb96ff0067bdee1aa153978e4e0/packages/webpack) directly.

Reply:
Yes I shed a tear when I saw this was async. We can def explore tapping into the internals, but it seems better to use their public API, at least for now, where the goal is stability. Can definitely dig into this more later, or even make a PR to nuxt (we are good friends with them).